diff -Naur /tmp/apple/portage/otool/main.c /tmp/apple/patched/otool/main.c
--- /tmp/apple/portage/otool/main.c	2014-02-16 10:35:47.860035245 -0800
+++ /tmp/apple/patched/otool/main.c	2013-09-23 05:14:34.000000000 -0700
@@ -282,7 +282,7 @@
 	progname = argv[0];
 	arch_flags = NULL;
 	narch_flags = 0;
-	all_archs = FALSE;
+	all_archs = TRUE;
 	use_member_syntax = TRUE;
 	llvm_mc = FALSE;
 
@@ -313,6 +313,7 @@
 		    all_archs = TRUE;
 		}
 		else{
+		    all_archs = FALSE;
 		    arch_flags = reallocate(arch_flags,
 			    (narch_flags + 1) * sizeof(struct arch_flag));
 		    if(get_arch_from_flag(argv[i+1],
diff -Naur /tmp/apple/portage/otool/print_objc.c /tmp/apple/patched/otool/print_objc.c
--- /tmp/apple/portage/otool/print_objc.c	2014-02-16 10:35:47.865035367 -0800
+++ /tmp/apple/patched/otool/print_objc.c	2013-09-23 05:14:34.000000000 -0700
@@ -31,7 +31,7 @@
 #include "stdio.h"
 #include "string.h"
 #include "mach-o/loader.h"
-#include "objc/objc-runtime.h"
+#include "objc/runtime.h"
 #include "stuff/allocate.h"
 #include "stuff/bytesex.h"
 #include "stuff/symbol.h"
@@ -635,7 +635,7 @@
 			printf("\n");
 		    printf("\t\t      isa 0x%08x", objc_class.isa);
 
-		    if(verbose && CLS_GETINFO(&objc_class, CLS_META)){
+		    if(verbose && objc_getMetaClass(objc_class.name)){
 			p = get_pointer(objc_class.isa, &left, objc_sections,
 					nobjc_sections, &cstring_section);
 			if(p != NULL)
@@ -674,9 +674,9 @@
 		    printf("\t\t     info 0x%08x",
 			   (unsigned int)objc_class.info);
 		    if(verbose){
-			if(CLS_GETINFO(&objc_class, CLS_CLASS))
+			if(objc_getClass(objc_class.name))
 			    printf(" CLS_CLASS\n");
-			else if(CLS_GETINFO(&objc_class, CLS_META))
+			else if(objc_getMetaClass(objc_class.name))
 			    printf(" CLS_META\n");
 			else
 			    printf("\n");
@@ -766,7 +766,7 @@
 			host_byte_sex, swapped, verbose) == FALSE)
 			printf(" (not in an " SEG_OBJC " section)\n");
 
-		    if(CLS_GETINFO((&objc_class), CLS_CLASS)){
+		    if(objc_getClass(objc_class.name)){
 			printf("\tMeta Class");
 			if(get_objc_class((uint32_t)objc_class.isa,
 			     &objc_class, &trunc, objc_sections, nobjc_sections,
