diff -Naur /tmp/apple/portage/ar/archive.c /tmp/apple/patched/ar/archive.c
--- /tmp/apple/portage/ar/archive.c	2014-02-16 10:35:47.895036094 -0800
+++ /tmp/apple/patched/ar/archive.c	2013-09-23 05:14:34.000000000 -0700
@@ -68,6 +68,7 @@
 
 #include <sys/param.h>
 #include <sys/stat.h>
+#include <sys/file.h>
 
 #include <ar.h>
 #include <dirent.h>
@@ -166,7 +167,6 @@
 		/* Locking seems to not be working */
 		case ENOTSUP:
 		case EHOSTUNREACH:
-		case EBADRPC:
 		default:
 			/* Filesystem does not support locking */
 			break;
@@ -321,7 +321,7 @@
 		 * things for exact binary equality.
 		 */
 		if (getenv("ZERO_AR_DATE") == NULL)
-			tv_sec = (long int)sb->st_mtimespec.tv_sec;
+			tv_sec = (long int)sb->st_mtime;
 		else
 			tv_sec = (long int)0;
 
diff -Naur /tmp/apple/portage/ar/misc.c /tmp/apple/patched/ar/misc.c
--- /tmp/apple/portage/ar/misc.c	2014-02-16 10:35:47.887035900 -0800
+++ /tmp/apple/patched/ar/misc.c	2013-09-23 05:14:34.000000000 -0700
@@ -164,7 +164,7 @@
 badfmt()
 {
 
-	errno = EFTYPE;
+	errno = EINVAL;
 	err(1, "%s", archive);
 }
 
